pkg/capi2mapi/aws.go

```diff
@@ -0,0 +1,432 @@
+/*
+Copyright 2024 Red Hat, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+	http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+package capi2mapi
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"strings"
+
+	mapiv1 "github.com/openshift/api/machine/v1beta1"
+	"gopkg.in/yaml.v2"
+
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	utilerrors "k8s.io/apimachinery/pkg/util/errors"
+	"k8s.io/utils/ptr"
+	capav1 "sigs.k8s.io/cluster-api-provider-aws/v2/api/v1beta2"
+	capiv1 "sigs.k8s.io/cluster-api/api/v1beta1"
+)
+
+var (
+	errCAPIMachineAWSMachineTemplateAWSClusterCannotBeNil = errors.New("provided Machine and AWSMachineTemplate and AWSCluster can not be nil")
+	errUnsupportedCAPATenancy                             = errors.New("unable to convert unsupported CAPA Tenancy")
```

Comment:
CAPA tenancy? Probably best to make it more about how the user would view this. We are unable to convert because this is not supported in Machine API, so maybe a message about field not supported in legacy Machine API implementation is better?